File: Parciales/Punto_4/punto4.py

import sys
import time


def fibonacci(n: int) -> int:
    if n <= 1:
        return n
    return fibonacci(n - 1) + fibonacci(n - 2)


def main() -> None:
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 35

    t1 = time.perf_counter()
    resultado = fibonacci(n)
    t2 = time.perf_counter()

    ms = (t2 - t1) * 1000
    print(f"Fibonacci({n}) = {resultado}")
    print(f"Tiempo Python (ms): {ms:.3f}")


if __name__ == "__main__":
    main()

File: Parciales/Punto_4/punto4.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

long long fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main(int argc, char **argv) {
    int n = 35;
    if (argc > 1) {
        n = atoi(argv[1]);
    }

    struct timespec t1, t2;
    clock_gettime(CLOCK_MONOTONIC, &t1);
    long long resultado = fibonacci(n);
    clock_gettime(CLOCK_MONOTONIC, &t2);

    double ms = (t2.tv_sec - t1.tv_sec) * 1000.0 +
                (t2.tv_nsec - t1.tv_nsec) / 1000000.0;

    printf("Fibonacci(%d) = %lld\n", n, resultado);
    printf("Tiempo C (ms): %.3f\n", ms);

    return 0;
}
